proof/invariant-abstract/RISCV64/ArchKernelInit_AI.thy

```diff
@@ -208,22 +209,59 @@ lemma pt_walk_init_A_st[simp]:
                    is_aligned_pt_slot_offset_pte global_pte_def)
   done
 
+(* FIXME MK: move to Word_Lib *)
+lemma word_and_mask_shift_eq_0:
+  "n \<le> m \<Longrightarrow> x && mask n >> m = 0"
+  by word_eqI
+
+(* FIXME MK: move to Word_Lib *)
+lemma word_mask_and_neg_shift_eq_0:
+  "n < m \<Longrightarrow> - (1 << m) && mask n = 0"
+  by (metis NOT_mask_AND_mask add.inverse_neutral leD less_imp_le mask_AND_NOT_mask mask_eqs(10)
+            shiftl_1 t2n_mask_eq_if)
+
+(* Since the bottom 30 bits don't matter for 1GB pages, kernelELFPAddrBase and physBase don't have
+   any influence on the index in the page table, i.e. the virtual address kernel_elf_base is always
+   within the same 1GB page. *)
+lemma elf_index_value:
```

Comment:
don't we call these `_def'`? haven't seen `_value` much

Reply:
Under `_def'` I would usually expect a different abstract representation and I wanted to make a distinction in the name to indicate you get a number when you use this (which we usually don't do). I thought I had seen `_val` or `_value` as a thing before, but it does indeed not seem to be mentioned anywhere.

If it's confusing, I'm fine with renaming to `def'`

Reply:
Hmm, def' is used for both values and different abstract representation. On the one hand, you're right, on the other, n+1 flavoured soup. I'm fine keeping `_value` and expect the next person who needs to do this to go with `_val`, so we get the full set.

Reply:
Happy to oblige and call the next one `_val` ;-)